Gems/Atom/Feature/Common/Code/Include/Atom/Feature/Utils/FrameCaptureBus.h

```diff
@@ -43,45 +45,60 @@ namespace AZ
+            //! @param imagePath The output file path. 
             //! @param passHierarchy For finding the pass by using a pass hierarchy filter. Check PassFilter::CreateWithPassHierarchy() function for detail
             //! @param slotName Name of the pass's slot. The attachment bound to this slot will be captured.
             //! @param option Only valid for an InputOutput attachment. Use PassAttachmentReadbackOption::Input to capture the input state
             //!               and use PassAttachmentReadbackOption::Output to capture the output state
-            //! @param outputFilePath The output file path. 
             //! @return value is the frame capture Id, on failure it will return InvalidFrameCaptureId
-            virtual FrameCaptureId CapturePassAttachment(const AZStd::vector<AZStd::string>& passHierarchy, const AZStd::string& slotName
-                , const AZStd::string& outputFilePath, RPI::PassAttachmentReadbackOption option) = 0;
+            virtual AZ::Outcome<FrameCaptureId, FrameCaptureError> CapturePassAttachment(
+                const AZStd::string& imagePath,
+                const AZStd::vector<AZStd::string>& passHierarchy,
+                const AZStd::string& slotName,
+                RPI::PassAttachmentReadbackOption option) = 0;
 
             //! Similar to CapturePassAttachment. But instead of saving the read back result to a file, it will call the callback function provide
             //! in the input when callback is finished
+            //! @param callback function to call when the data is ready to read
             //! @param passHierarchy For finding the pass by using a pass hierarchy filter. Check PassFilter::CreateWithPassHierarchy() function for detail
             //! @param slotName Name of the pass's slot. The attachment bound to this slot will be captured.
-            //! @param callback function to call when the data is ready to read
             //! @param option Only valid for an InputOutput attachment. Use PassAttachmentReadbackOption::Input to capture the input state
             //!               and use PassAttachmentReadbackOption::Output to capture the output state
             //! @return value is the frame capture Id, on failure it will return InvalidFrameCaptureId
-            virtual FrameCaptureId CapturePassAttachmentWithCallback(const AZStd::vector<AZStd::string>& passHierarchy, const AZStd::string& slotName
-                , RPI::AttachmentReadback::CallbackFunction callback, RPI::PassAttachmentReadbackOption option) = 0;
+            virtual AZ::Outcome<FrameCaptureId, FrameCaptureError> CapturePassAttachmentWithCallback(
+                RPI::AttachmentReadback::CallbackFunction callback,
+                const AZStd::vector<AZStd::string>& passHierarchy,
+                const AZStd::string& slotName,
+                RPI::PassAttachmentReadbackOption option) = 0;
```

Comment:
why re-order the arguments?

Reply:
The most critical arguments should be the callback, while `CapturePassAttachment` focuses on output path. It is kind of weird to have the argument we care in between 3 reference arguments.